src/main/java/org/broadinstitute/hellbender/tools/walkers/bqsr/BaseRecalibrator.java

```diff
@@ -176,10 +180,10 @@ public void onTraversalStart() {
         if (RAC.knownSites.isEmpty() && !RAC.RUN_WITHOUT_DBSNP) // Warn the user if no dbSNP file or other variant mask was specified
             throw new UserException.CommandLineException(NO_DBSNP_EXCEPTION);
 
-        requestedCovariates = getCovariatesArray();
+        covariates = new StandardCovariateList();
 
         logger.info("The covariates being used here: ");
-        for (Covariate cov : requestedCovariates) { // list all the covariates being used
+        for (Covariate cov : covariates) { // list all the covariates being used
             logger.info("\t" + cov.getClass().getSimpleName());
             cov.initialize(RAC); // initialize any covariate member variables using the shared argument collection
```

Comment:
This is an excellent point. I think the whole covariate abstraction is somewhat broken. In particular, what is a covariate? It seems reasonable to say that a covariate is a function from reads to arrays of ints (function is parameterized somewhat by eg context size etc). The ReadGroupCovariate causes a problem here because its encoding scheme is essentially hard-wired to one specific BAM file - even worse than that, its hardwired to a specific pass over the one BAM file (ie depends on the order of processing the reads).